promql/engine.go

```diff
@@ -1039,12 +1039,26 @@ type groupedAggregation struct {
 }
 
 // aggregation evaluates an aggregation operation on a vector.
-func (ev *evaluator) aggregation(op itemType, grouping model.LabelNames, keepExtra bool, vec vector) vector {
+func (ev *evaluator) aggregation(op itemType, grouping model.LabelNames, without bool, keepExtra bool, vec vector) vector {
 
 	result := map[uint64]*groupedAggregation{}
 
 	for _, sample := range vec {
-		groupingKey := model.SignatureForLabels(sample.Metric.Metric, grouping...)
+		var withoutMetric metric.Metric
+		if without {
+			withoutMetric = *sample.Metric.Copy()
```

Comment:
For `by` we build the metric in the if condition above. Doing it here for without makes things inconsistent and hard to read.

Also `metric.Metric` is already the copy-on-write version. So this line can essentially be dropped.